ebos/eclwriter.hh

```diff
@@ -588,7 +684,8 @@ private:
         // callback to eclIO serial writeTimeStep method
         void run()
         {
-            eclIO_.writeTimeStep(reportStepNum_,
+            eclIO_.writeTimeStep(summaryState,
```

Comment:
This is the only serious defect I think in this code. I believe at this point you are passing (a reference to) the SummaryState object owned by the vanguard. That will change in the next call to writeOutput(), which may happen before this thread is done. To be safe, only constant objects may be passed by reference to EclWriteTasklet. The EclipseIO object itself is passed by reference, but I believe that is *only* changed inside this call, so the pipelineing of the tasklets prevents this from being a problem.

Either we must find a way to avoid passing the SummaryState, or a copy must be passed to the EclWriteTasklet. What is the state used for inside writeTimeStep().

Reply:
Many good points - I am working on an update.

Reply:
I appreciate the changes, in particular splitting the long function and moving the eval() out of the tasklet. However I still think this code is buggy (although not introduced in this PR as I thought) for the reasons outlined above?

Reply:
Ehhh - I might misunderand, bu I think you are wrong. The tasklet now takes a copy.

Reply:
Line 664 says it's a reference. Where is the copy?

Also noticed new nitpick: we now have code mixing trailing underscores and no trailing underscores. Should be fixed.

Reply:
> we now have code mixing trailing underscores and no trailing underscores. Should be fixed.

OK - now I have been sitting on my hands for about three hours, and body temperatur has fallen below 38&deg; C and I am ready to start sprinkling the code with underscores.

Reply:
> Line 664 says it's a reference. Where is the copy?

You are right - it was only in my head I had changed that to a copy; *now* it is a copy though.

Reply:
> OK - now I have been sitting on my hands for about three hours - at a high hourly rate - body temperature has fallen below 38° C and I am ready to start sprinkling the code with underscores.

It was only a nitpick, it was not my intention to cause bodily harm... But I think you would agree that we should keep consistent within each class.

Reply:
> It was only a nitpick,

Well - either you write a comment and expect it to be adhered to, or you let it be.  

> it was not my intention to cause bodily harm... But I think you would agree that we should keep consistent within each class.

Of course I was trying to make a funny point; I did not sit on my hands while cooling off, instead I went pacing around the house. But seriously - and this you can quote me on: "The argument of consistency is consistently brought up when it aligns with the general preference of the person bringing it up. A bit like business leaders who argue for a *fair* tax regime - when what they really want is zero tax just for them." Of course you are taking more than your fair share of Sintef flak now; anyway underscores have been introduced on the `SummaryState` member.

Reply:
don't claim this is a sintef thing! i work in sintef and i completely disagree. trailing underscores are evil eviil things. the only sane thing to do is to let members  have a m_ prefix :trollface:

Reply:
I don't know if you are serious now - and I will anyway not spam any further on the topic, but just for the record I consider the `m_` prefix high on the top ten list of my professional blunders. My current rage on the matter is using `this->` consistently #IalsoLikePython